### PR TITLE
Add Collected Curios to Tachiyomi extensions.

### LINKED ISSUE #17670: [Source Request] Collected Curios
### Source name

Collected Curios

### Source link

https://www.collectedcurios.com/

### Source language

English

### Other details

Requesting Collected Curios by Jolly jack as in #8475

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Tomoka1**
Adds Collected Curios to Tachiyomi extensions.
The site is quite splattered so further work is still need in adding more categories.
Main interests are available though, that's why i am submitting this merge request.

Closes #17670

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

**Comment by alessandrojean** [comment]
Can you use the icon generator tool linked at the CONTRIBUTING.md guide? This one seems to be missing the shadow.

File: src/en/collectedcurios/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest package="eu.kanade.tachiyomi.extension" />
```
Comment (arkon) [inline src/en/collectedcurios/AndroidManifest.xml:2]:
```suggestion
<manifest />
```

File: src/en/collectedcurios/build.gradle [hunk 2]
```diff
@@ -0,0 +1,12 @@
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlinx-serialization'
+
+ext {
+    extName = 'Collectedcurios by Jolly Jack'
+    pkgNameSuffix = 'en.collectedcurios'
+    extClass = '.Collectedcurios'
+    extVersionCode = 2
```
Comment (arkon) [inline src/en/collectedcurios/build.gradle:9]:
```suggestion
    extVersionCode = 1
```

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+                        title = "Battle Bunnies"
+                        artist = "Jolly Jack aka Phillip M Jackson"
+                        author = "Jolly Jack aka Phillip M Jackson"
+                        status = SManga.ONGOING
+                        url = "/battlebunnies.php"
+                        description = "Battle Bunnies webcomic."
+                        thumbnail_url = "https://www.collectedcurios.com/images/CC_2011_Battle_Bunnies_Button.jpg"
+                    },
+
+                    /*
```
Comment (arkon) [inline src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:51]: Remove if not needed?

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+                    },
+                    */
+                ),
+                false,
+            ),
+        )
+    }
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList): Observable<MangasPage> {
+        // return Observable.just(MangasPage(emptyList(), false))
```
Comment (arkon) [inline src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:69]: Remove if not needed?

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+        var chapterList = mutableListOf<SChapter>()
+
+        for (i in chapters?.downTo(1)!!) {
+            chapterList.add(
+                SChapter.create().apply {
+                    url = listOf(response.request.url, "?s=", i).joinToString("")
+                    name = listOf("Chapter", i).joinToString(" - ")
+                    chapter_number = i.toFloat()
+                    date_upload = 0L
```
Comment (arkon) [inline src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:97]: Why explicitly 0?

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+                    ?.attr("href")?.substringAfter("=")?.toInt()
+            }
+
+        var chapterList = mutableListOf<SChapter>()
+
+        for (i in chapters?.downTo(1)!!) {
+            chapterList.add(
+                SChapter.create().apply {
+                    url = listOf(response.request.url, "?s=", i).joinToString("")
+                    name = listOf("Chapter", i).joinToString(" - ")
```
Comment (arkon) [inline src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:95]: Seems like an overkill way to build strings. Just inline them.

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        }
+        return chapterList
+    }
+
+    override fun chapterListSelector() = throw Exception("Not used")
+
+    override fun chapterFromElement(element: Element) = SChapter.create().apply {
+        name = element.selectFirst(".w3-round")?.attr("value")!!.ifBlank { "date" }
+        date_upload = 0L
+    }
```
Comment (arkon) [inline src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:109]: Why explicitly 0?

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+                },
+            )
+        }
+        return chapterList
+    }
+
+    override fun chapterListSelector() = throw Exception("Not used")
+
+    override fun chapterFromElement(element: Element) = SChapter.create().apply {
+        name = element.selectFirst(".w3-round")?.attr("value")!!.ifBlank { "date" }
```
Comment (arkon) [inline src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:107]: Weird name ("date")?

**Review by arkon — CHANGES_REQUESTED** [review]

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt (hunk 5 above)
Comment (Tomoka1) [reply src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:97]: Copied from another extension. Looked like a default value.

Comment (Tomoka1) [reply src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:97]: Copied from another extension. Looked like a default value to me.

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt (hunk 3 above)
Comment (Tomoka1) [reply src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:51]: Will be needed in next version.

File: src/en/collectedcurios/AndroidManifest.xml (hunk 1 above)
Comment (Tomoka1) [reply src/en/collectedcurios/AndroidManifest.xml]: Copied from another extension. Seemed logical to me, but i found the other "Remove package names" commit.

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt (hunk 8 above)
Comment (Tomoka1) [reply src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:106]: Can probably be removed all together.

**Comment by Tomoka1** [comment]
Sorry, this is my first actual code review. I hope i didn't botch it by resolving everything now. I addressed most of Arkons issues now. Some i am keeping as reminder.

File: src/en/collectedcurios/build.gradle [hunk 9]
```diff
@@ -0,0 +1,12 @@
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlinx-serialization'
+
+ext {
+    extName = 'Collectedcurios by Jolly Jack'
```
Comment (alessandrojean) [inline src/en/collectedcurios/build.gradle:6]: Can name it like the source itself.

```suggestion
    extName = 'Collected Curios'
```

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt (hunk 8 above)
Comment (alessandrojean) [reply src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:106]:
```suggestion
        name = element.selectFirst(".w3-round")?.attr("value") ?: "Chapter"
```

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+        date_upload = 0L
```
Comment (alessandrojean) [inline src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:107]: Doesn't need to set if you don't have the date.

```suggestion
```

File: src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+            ccResponse.contains("spiderandscorpion") -> {
+                val imageName = response.asJsoup().selectFirst("img[id=strip]")?.attr("src")
+                return listOf(baseUrl, imageName).joinToString("/")
+            }
+
+            else -> {
+                return listOf("", "").joinToString("/")
+            }
+        }
+    }
```
Comment (alessandrojean) [inline src/en/collectedcurios/src/eu/kanade/tachiyomi/extension/en/collectedcurios/Collectedcurios.kt:136]:
```suggestion
    override fun imageUrlParse(response: Response): String {
        val url = response.request.url.toString()
        val document = response.asJsoup()
        
        return when {
            url.contains("sequentialart") ->
                document.selectFirst(".w3-image")!!.absUrl("src")
            url.contains("battlebunnies") || url.contains("spiderandscorpion") ->
                document.selectFirst("#strip")!!.absUrl("src")
            else -> throw Exception("Could not find the image")
        }
    }
```

**Review by alessandrojean — CHANGES_REQUESTED** [review]

**Pull request merged**